packages/rocketchat-api/server/routes.coffee

```diff
@@ -185,5 +196,195 @@ RocketChat.API.v1.addRoute 'groups.create', authRequired: true,
 			return RocketChat.API.v1.failure e.name + ': ' + e.message
 
 		return RocketChat.API.v1.success
-			group: RocketChat.models.Rooms.findOne({_id: id.rid})
+			group: RocketChat.models.Rooms.findOne({_id: id.rid})		
+
+
+#list All Private Groups		
+RocketChat.API.v1.addRoute 'privateRooms.list', authRequired: true,
+	get: ->
+		try
+			this.response.setTimeout (1000)
+			rooms = RocketChat.models.Rooms.findByType('p', { sort: { msgs:-1 } }).fetch()
+			#status: 'success', rooms: rooms
+		catch e
+			return RocketChat.API.v1.failure e.name + ': ' + e.message
+		return RocketChat.API.v1.success
+			list: rooms
+			
+			
+#list All Direct Groups		
+RocketChat.API.v1.addRoute 'directRoom.list', authRequired: true,
+	get: ->
+		try
+			this.response.setTimeout (1000)
+			rooms = RocketChat.models.Rooms.findByType('d', { sort: { msgs:-1 } }).fetch()
+		catch e
+			return RocketChat.API.v1.failure e.name + ': ' + e.message
+		return RocketChat.API.v1.success
+			list: rooms			
+			
+			
+# addUser to a channel/private group
+RocketChat.API.v1.addRoute 'addUser', authRequired: true,
+	post: ->
+		if RocketChat.authz.hasPermission(@userId, 'bulk-create-c')
```

Comment:
You don't need to verify this permission here, there is no relation with `bulk-create-c` and add an user to a room, the method `addUserToRoom` is doing the permission verification for you